services/QuillLMS/app/lib/snapshots/cache_keys.rb

```diff
@@ -2,17 +2,17 @@
 
 module Snapshots
   class CacheKeys
-    def self.generate_key(query, timeframe_name, current_start, current_end, school_ids, additional_filters: {})
-      root_key(query, timeframe_name, current_start, current_end, school_ids)
+    def self.generate_key(report, query, timeframe_name, current_start, current_end, school_ids, additional_filters: {})
```

Comment:
I like your decision to separate the concern of key generation from the concern of cache fetching.